components/profile/mem.rs

```diff
@@ -35,16 +41,25 @@ impl Profiler {
     pub fn create(period: Option<f64>) -> ProfilerChan {
         let (chan, port) = ipc::channel().unwrap();
 
+        let notifier = Arc::new((Mutex::new(false), Condvar::new()));
+        let notifier2 = Arc::clone(&notifier);
+
         // Create the timer thread if a period was provided.
         if let Some(period) = period {
             let chan = chan.clone();
             thread::Builder::new()
                 .name("MemoryProfTimer".to_owned())
                 .spawn(move || loop {
                     thread::sleep(Duration::from_secs_f64(period));
+                    let (mutex, cvar) = &*notifier;
+                    let mut done = mutex.lock();
+                    *done = false;
                     if chan.send(ProfilerMsg::Print).is_err() {
                         break;
                     }
+                    if !*done {
```

Comment:
This condition is always true, right? The mutex remains locked the whole time and we assigned a false value to the boolean.

Reply:
Sorry, I didn't mean to.imply that it wouldn't work, only that the `if` that I highlighted is unnecessary since it cannot evaluate to true. Yes, wait does drop the mutex until it is reawakened.

Reply:
Eh, it's not an argument that I need to win :)